common/src/test/kotlin/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketCoreImplTest.kt

```diff
@@ -266,7 +288,71 @@ connect()
         assertTrue(result)
         // auth
         coVerify(exactly = 1) { mockConnection.send(any<String>()) }
-        assertSame(WebSocketState.ACTIVE, webSocketCore.getConnectionState())
+        assertEquals(WebSocketState.ACTIVE, webSocketCore.getConnectionState())
+    }
+
+    @Test
+    fun `Given an active connection When URL changes Then it cancels the connection`() = runTest {
```

Comment:
```suggestion
    fun `Given an active connection When URL changes to a different URL Then it cancels the connection`() = runTest {
```

Make the difference with the second test more obvious